pyhive/hive.py

```diff
@@ -299,23 +302,29 @@ def fetch_logs(self):
         if self._state == self._STATE_NONE:
             raise ProgrammingError("No query yet")
 
-        logs = []
-        while True:
-            req = ttypes.TFetchResultsReq(
-                operationHandle=self._operationHandle,
-                orientation=ttypes.TFetchOrientation.FETCH_NEXT,
-                maxRows=self.arraysize,
-                fetchType=1  # 0: results, 1: logs
-            )
-            response = self._connection.client.FetchResults(req)
-            _check_status(response)
-            assert not response.results.rows, 'expected data in columnar format'
-            assert len(response.results.columns) == 1, response.results.columns
-            new_logs = _unwrap_column(response.results.columns[0])
-            logs += new_logs
+        try:  # Older Hive instances require logs to be retrieved using GetLog
```

Comment:
- Perhaps make the modern version first, so users don't have to pay a performance penalty? Or if there's a way to detect this in the protocol, that would be better.
- could you document which versions this is for?